plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/image/deprecated.tsx

```diff
@@ -1,66 +1,51 @@
 /**
  * External dependencies
  */
-import clsx from 'clsx';
 import { createBlock } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import metadata from './block.json';
 import { BlockAttributesV1 } from './types';
+import save from '../save';
 
-// In v2, we removed the `showSaleBadge` attribute and converted it to an inner block.
+// In v2, we're migrating the `showSaleBadge` attribute to an inner block.
 const v1 = {
-	attributes: {
-		...metadata.attributes,
-		showSaleBadge: {
-			type: 'boolean',
-			default: true,
-		},
-		saleBadgeAlign: {
-			type: 'string',
-			default: 'right',
-		},
+	save,
+	attributes: metadata.attributes,
+	isEligible: ( { showSaleBadge }: BlockAttributesV1 ) => {
+		// If the block is pristine, it doesn't have a showSaleBadge attribute
+		// but it is `true` by default.
+		const trueByDefault = showSaleBadge === undefined;
+
+		// If the block was edited, it will have a showSaleBadge attribute
+		// that we should respect.
+		const trueAfterEdit = showSaleBadge === true;
+
+		return trueByDefault || trueAfterEdit;
```

Comment:
Another nitpick, but this code is repeated and I know it's only twice, but maybe at this point making it into a small function by itself could improve readability.

Reply:
Done https://github.com/woocommerce/woocommerce/pull/57889/commits/d172eef438ac47d24d7a6b5a15e736b340241f52